lib/watch-files.js

```diff
@@ -29,11 +54,12 @@ module.exports = (nameOfDirectory) => {
         const parsed = path.parse(typeFile);
         pattern = parsed.base;
         command.type = parsed.dir.replace(path.sep, '');
-      } else {
-        pattern = file.replace(`${nameOfDirectory}${path.sep}`, `*${path.sep}`);
       }
-      logResult('change', `occurred on ${file}, kicking off upload`, 'magentaBright');
-      await uploader(pattern, command, config);
+      logResult('change', `occurred on ${file}, queueing upload`, 'magentaBright');
+
+      if (queueData.has(file)) { queue.splice(queue.indexOf(file)); }
```

Comment:
I think this should be looking at the queue because you could have already popped off this file from the queue and it's currently uploading but your queueData still has a reference to the file in it. and it's going to cause this to pop the end of the queue off.

Reply:
Good point. I was just thinking about which is faster.